crates/autopilot/src/domain/settlement/encoded/mod.rs

```diff
@@ -0,0 +1,249 @@
+//! This module defines Settlement in an onchain compatible format.
+
+use {
+    crate::{
+        boundary,
+        domain::{
+            self,
+            auction::{self, order},
+            eth,
+        },
+    },
+    ethcontract::{common::FunctionExt, tokens::Tokenize, Address, Bytes, U256},
+};
+
+pub mod tokenized;
+
+/// Settlement in an encoded format, as expected by the settlement contract
+/// `settle` function.
+///
+/// Type safe representation of the settlement transaction calldata.
+#[derive(Debug)]
+pub struct Encoded {
```

Comment:
I think the type itself should probably be called settlement. I'm not sure about the module path. It's not really encoded (in fact we are decoding it from calldata), but maybe a more neutral name describing the raw byte representation of the settlement from on chain would be good (maybe `coded`).

Reply:
renamed